### ISSUE TITLE
PB740: Rotated screen stalls since v2020.11

### ISSUE THREAD

**Issue by thomasjfox** [koreader/koreader#7010]
Hi ezdiy,

after updating from koreader v2020.10.1 to v2020.11 on a Pocketbook InkPad 3  (PB740), the screen updates erratically in landscape mode. Portrait mode is fine.

At first I thought the touch events are not registered, but touching several times *sometimes* updates the screen.

There's a workaround though: Enable the fastest eink update speed 3, then it's working as in 2020.10.1.

I think it might be related to 

```
commit 5a86443eb1d9069715f1bca98b33d9f46ec2b068
Refs: v2020.10.1-15-g5a86443e
Author:     ezdiy
AuthorDate: Thu Oct 22 13:23:09 2020 +0200

    Pocketbook: waveform presets (#6794)
```
#6794 was part of v2020.11.

Another related change: https://github.com/koreader/koreader/pull/6734
"Pocketbook: Fix silly rotation bug on PB740"

Any idea what might be going on?

This is a non-rooted device, I think ^^

Thanks for the great work for the Pocketbook devices, ezdiy!
It really made koreader fun to use on the PB740.

Cheers,
Thomas

**Comment by NiLuJe** [comment]
IIRC, it might require switching to compat waveform modes in Gear > Screen > E-ink settings >

**Comment by thomasjfox** [comment]
Might be relevant: This happens on FW 5.20.1010.

I just found out that FW 6.1.1229 has been released on 2020-11-27 :)

**Comment by thomasjfox** [comment]
> IIRC, it might require switching to compat waveform modes in Gear > Screen > E-ink settings >

which is the odd part: The "Level 0: high quality, slowest" doesn't work, but "Level 3: low quality, fastest" does. I thought the compat one is level 0. Also portrait mode is not affected.

Thanks for the suggestion.

**Comment by NiLuJe** [comment]
Might be because the "fast" preset disables hardware rotation ;).

**Comment by thomasjfox** [comment]
Upgrading to the latest "just released" InkPad firmware solved the issue in landscape mode. Great! The device also starts faster as a little Christmas present.

I'm wondering if there's a way to detect the "previous stable" firmware version and warn the user? I'm tech savy and it took me a few automated koreader up- and downgrades to pinpoint the issue. For other users it might feel as "the display just freezes".

**Comment by NiLuJe** [comment]
There's *way too many* combinations possible on PB to make any semblance of sense out of things kinda tricky.

But, yeah, I'd imagine there's an InkView function that spits out the FW version.

We don't really have many people familiar with the PB platform, hence the weirdness. (@ezdiy is currently away; @pazos now has access to one, but is generally busy dealing with Android being the absolute worst, so piling on "PB being terrible" on top of it may affect his sanity ^^).

**Comment by Frenzie** [comment]
> generally busy dealing with Android being the absolute worst

And I greatly appreciate that. :+1:

**Comment by pazos** [comment]
I recovered my PB recently (It was locked down far away from me). I can confirm that waveform levels screw my device (PB HD3). Level 0 is ok but **slowwww** and has the rotated infomessage on start. The rest of levels are not usable as they seem to aply fast wf to everything.

Things worked much better on my device with https://github.com/koreader/koreader-base/pull/1181 applied, which was the last time I tinkered with the device. Sadly I didn't followed changes since then and I'm not sure what bits need to be reverted to get the same results now.

I can make a video vs my cervantes if you want. It is not fair as the cervantes has faster eink refreshes that any other device that I played with.
